src/types.ts

```diff
@@ -1,6 +1,6 @@
 export interface ObsidianGitSettings {
     commitMessage: string;
-    autoCommitMessage: string;
+    autoCommitMessage?: string; // possibly undefined for settings migration
```

Comment:
I would rather make this non-undefined again as settings migration runs at the very beginning. So it's everywhere used non-undefined.

Reply:
I see... I think we can just omit `autoCommitMessage` in `DEFAULT_SETTINGA`.

Reply:
I added string as default value instead of `undefined`.
1d6a92e